src/HexagonOptimize.cpp

```diff
@@ -112,11 +112,16 @@ struct Pattern {
         ExactLog2Op1 = 1 << 3, // Replace operand 1 with its log base 2, if the log base 2 is exact.
         ExactLog2Op2 = 1 << 4, // Save as above, but for operand 2.
 
+        FirstExactLog2Op = 1,   // FirstExactLog2Op and NumExactLog2Op ensure that we check only op1 and op2
+        NumExactLog2Op = 2,     // for ExactLog2Op
```

Comment:
I think the code below would be easier (less verbose) if this were BeginExactLog2Op and EndExactLog2Op (i.e. container iterator style). That way the loop would read:

    for (size_t i = Pattern::BeginExactLog2Op; i < Pattern::EndExactLog2Op && is_match; i++) {

Also, note in the comment that these are not flags. It took me a bit to figure out that these were not flags, and the comment is a bit confusing. I would rewrite the comment to say something like:

    These are not flags, they identify the first and last valid operands for ExactLog2Op.